libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -227,6 +227,20 @@ impl NotificationHandler {
     }
 }
 
+impl Iterator for NotificationHandler {
```

Comment:
I think the more appropriate trait for this use case is to implement `IntoIterator`.  See https://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html#impl-IntoIterator.  It should be pretty simple, as you can just return the Receiver.

Reply:
Could you elaborate a bit? I'm sure why this would be more appropriate.

Reply:
I guess it's a question of whether or not a NotificationHandler should _be_ an iterator or can be transformed _into_ an iterator.

Reply:
I think it makes more sense as an Iterator, since it doesn't have much purpose otherwise, and having it implement `IntoIterator` instead just seems to add an extra step (calling `notifier.next()` vs calling `notifier.into_iter().next()`.

Reply:
I'm not sure I agree with that usage, though.  By hiding the blocking version behind `Iterator::next`, it's not exactly documented.  It should be its own method on Notifier, and the iterator provide just calls it.

Reply:
Ok. So you're saying don't remove `listen`? Because what you're describing is what `listen` was. Should we even have an iterator then?

Reply:
We can still have an iterator, but `listen` should be part of the public API.  Iterator is still a nice convenience, so it can be both.

Reply:
Changed `try_next` to `try_listen`, and add listen back.